ir/type.cpp

```diff
@@ -664,6 +668,20 @@ VectorType::VectorType(string &&name, unsigned elements, Type &elementTy)
   }
 }
 
+StateValue VectorType::aggregateVals(const vector<StateValue> &vals) const {
+  assert(defined);
+  assert(vals.size() == elements);
+
+  expr val;
+  expr not_poison;
+  for (unsigned idx = 0; idx < elements; ++idx) {
+    auto [v, vs] = children[idx]->toBV(vals[idx]);
+    val = idx == 0 ? v : val.concat(v);
```

Comment:
StateValue now supports concat. You can do this in one go :)
See AggregateConst::toSMT().  Actually that code should be replaced with a call to this.

Reply:
BTW, I think this method can be hoisted to the AggregateType class. The extract is also there.

Reply:
Fixed, these makes code extremely clean!